testsuite/httpx/__init__.py

```diff
@@ -65,6 +65,15 @@ def __str__(self):
         return f"Result[error={self.error}]"
 
 
+class ResultList(list):
+    """List-like object for Result"""
+
+    def assert_all(self, status_code=None):
+        """Assert all responses that contain certain status code"""
+        for request in self:
+            assert request.status_code == status_code, f"Rate Limited resource unexpectedly rejected request: {request}"
```

Comment:
I would improve the error message wording and to also include index of the request that is wrong